core/domain/activity_jobs_one_off_test.py

```diff
@@ -2078,3 +2081,412 @@ def test_audit_multiple_rights_snapshots(self):
                 ['topic-length-0', 1]
             ]
         )
+
+
+class ValidateSnapshotMetadataModelsJobTests(test_utils.GenericTestBase):
+    ALBERT_EMAIL = 'albert@example.com'
+    ALBERT_NAME = 'albert'
+
+    EXP_ID = 'exp_id0'
+    COLLECTION_ID = 'collection_id0'
+    QUESTION_ID = 'question_id0'
+    SKILL_ID = 'skill_id0'
+    STORY_ID = 'story_id0'
+    TOPIC_ID = 'topic_id0'
+    # The subtopic snapshot ID is in the format
+    # '<topicId>-<subtopicNum>-<version>'.
+    SUBTOPIC_ID = 'topic_id0-1'
+    TOPIC_RIGHTS_ID = 'topic_rights_id0'
+    DUMMY_COMMIT_CMDS = [
+        {
+            'cmd': 'some_command',
+            'other_field': 'test'
+        }, {
+            'cmd': 'some_other_command',
+            'other_field': 'test',
+            'different_field': 'test'
+        }
+    ]
+
+    def setUp(self):
+        super(ValidateSnapshotMetadataModelsJobTests, self).setUp()
+
+        # Setup user who will own the test explorations.
+        self.signup(self.ALBERT_EMAIL, self.ALBERT_NAME)
+        self.albert_id = self.get_user_id_from_email(self.ALBERT_EMAIL)
+        self.process_and_flush_pending_tasks()
+
+    def _run_one_off_job(self):
+        """Runs the one-off MapReduce job."""
+        job_id = (
+            activity_jobs_one_off.ValidateSnapshotMetadataModelsJob.create_new()) # pylint: disable=line-too-long
+        activity_jobs_one_off.ValidateSnapshotMetadataModelsJob.enqueue(job_id)
+        self.assertEqual(
+            self.count_jobs_in_mapreduce_taskqueue(
+                taskqueue_services.QUEUE_NAME_ONE_OFF_JOBS), 1)
+        self.process_and_flush_pending_mapreduce_tasks()
+        stringified_output = (
+            activity_jobs_one_off.ValidateSnapshotMetadataModelsJob
+            .get_output(job_id))
+
+        eval_output = [ast.literal_eval(stringified_item) for
+                       stringified_item in stringified_output]
+        return eval_output
+
+    def test_correct_collection_models(self):
+        rights_manager.create_new_collection_rights(
+            self.COLLECTION_ID, self.albert_id)
+
+        collection_model = collection_models.CollectionModel(
+            id=self.COLLECTION_ID,
+            category='category',
+            title='title',
+            objective='objective',
+            collection_contents={
+                'nodes': {}
+            },
+        )
+        collection_model.commit(
+            self.albert_id, 'collection model created', self.DUMMY_COMMIT_CMDS)
+
+        actual_output = self._run_one_off_job()
+
+        expected_output = [
+            ['FOUND PARENT MODEL-CollectionRightsSnapshotMetadataModel', 1],
+            ['FOUND COMMIT LOGS-CollectionSnapshotMetadataModel', 1],
+            ['FOUND PARENT MODEL-CollectionSnapshotMetadataModel', 1],
+            [
+                'COMMIT LOGS SHOULD NOT EXISTS-' +
```

Comment:
EXISTS --> EXIST

Have a space on each side of the "-", for all messages.

Reply:
Done.